.github/scripts/generate_binary_build_matrix.py

```diff
@@ -372,28 +372,27 @@ def generate_wheels_matrix(
                             "use_split_build": use_split_build,
                         }
                     )
-            else:
-                ret.append(
-                    {
-                        "python_version": python_version,
-                        "gpu_arch_type": gpu_arch_type,
-                        "gpu_arch_version": gpu_arch_version,
-                        "desired_cuda": translate_desired_cuda(
-                            gpu_arch_type, gpu_arch_version
-                        ),
-                        "devtoolset": "cxx11-abi"
-                        if arch_version == "cpu-cxx11-abi"
-                        else "",
-                        "container_image": WHEEL_CONTAINER_IMAGES[arch_version],
-                        "package_type": package_type,
-                        "build_name": f"{package_type}-py{python_version}-{gpu_arch_type}{gpu_arch_version}".replace(
-                            ".", "_"
-                        ),
-                        "pytorch_extra_install_requirements":
-                        PYTORCH_EXTRA_INSTALL_REQUIREMENTS["12.1"]  # fmt: skip
-                        if os != "linux" else "",
-                    }
-                )
+                else:
+                    ret.append(
```

Comment:
This is mostly the same as the `ret.append` contents above. Why not merged into a single dict?  You can conditionally add `pytorch_extra_install_requirements` when needed, and then return the final dict